pkg/kubelet/status/manager.go

```diff
@@ -396,7 +396,12 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 				glog.V(3).Infof("Pod %q is terminated, but some containers are still running", format.Pod(pod))
 				return
 			}
-			if err := m.kubeClient.Core().Pods(pod.Namespace).Delete(pod.Name, api.NewDeleteOptions(0)); err == nil {
+			// The delete options guaranteed that:
+			//   1. The pod should be deleted immediately
+			//   2. Only the pod with the same uid should be deleted. See #25169
```

Comment:
This is obviously the right thing to do. Don't think we need to link the issue :-)
maybe change the comment to "Use the pod UID as the precondition for deletion to prevent deleting a newly created pod with the same name and namespace"